manifests/fact.pp

```diff
@@ -34,18 +34,38 @@
   $match = "^${name}=\\S.*$"
 
   if $facts_file != $facter::facts_file {
-    file { "facts_file_${name}":
-      ensure => file,
-      path   => $facts_file_path,
-      owner  => $facter::facts_file_owner,
-      group  => $facter::facts_file_group,
-      mode   => $facter::facts_file_mode,
+    if $facter::facts_file_purge {
```

Comment:
Why not use concat all the time instead of file line and reduce the logic here?

Reply:
I've updated to only use concat